// @internal/prisma-old/prisma/client/index-browser.js

Object.defineProperty(exports, "__esModule", { value: true });

const {
  Decimal
} = require('./runtime/index-browser')


const Prisma = {}

exports.Prisma = Prisma

/**
 * Prisma Client JS version: 3.14.0
 * Query Engine version: 2b0c12756921c891fec4f68d9444e18c7d5d4a6a
 */
Prisma.prismaVersion = {
  client: "3.14.0",
  engine: "2b0c12756921c891fec4f68d9444e18c7d5d4a6a"
}

Prisma.PrismaClientKnownRequestError = () => {
  throw new Error(`PrismaClientKnownRequestError is unable to be run in the browser.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues`,
)};
Prisma.PrismaClientUnknownRequestError = () => {
  throw new Error(`PrismaClientUnknownRequestError is unable to be run in the browser.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues`,
)}
Prisma.PrismaClientRustPanicError = () => {
  throw new Error(`PrismaClientRustPanicError is unable to be run in the browser.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues`,
)}
Prisma.PrismaClientInitializationError = () => {
  throw new Error(`PrismaClientInitializationError is unable to be run in the browser.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues`,
)}
Prisma.PrismaClientValidationError = () => {
  throw new Error(`PrismaClientValidationError is unable to be run in the browser.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues`,
)}
Prisma.Decimal = Decimal

/**
 * Re-export of sql-template-tag
 */
Prisma.sql = () => {
  throw new Error(`sqltag is unable to be run in the browser.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues`,
)}
Prisma.empty = () => {
  throw new Error(`empty is unable to be run in the browser.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues`,
)}
Prisma.join = () => {
  throw new Error(`join is unable to be run in the browser.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues`,
)}
Prisma.raw = () => {
  throw new Error(`raw is unable to be run in the browser.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues`,
)}
Prisma.validator = () => (val) => val

/**
 * Shorthand utilities for JSON filtering
 */
Prisma.DbNull = 'DbNull'
Prisma.JsonNull = 'JsonNull'
Prisma.AnyNull = 'AnyNull'

/**
 * Enums
 */
// Based on
// https://github.com/microsoft/TypeScript/issues/3192#issuecomment-261720275
function makeEnum(x) { return x; }

exports.Prisma.UserScalarFieldEnum = makeEnum({
  id: 'id',
  email: 'email',
  emailVerified: 'emailVerified',
  password: 'password',
  tokenVersion: 'tokenVersion',
  disabled: 'disabled',
  isAdmin: 'isAdmin',
  googleId: 'googleId',
  stripeCustomerId: 'stripeCustomerId',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
});

exports.Prisma.ProfileScalarFieldEnum = makeEnum({
  uid: 'uid',
  displayName: 'displayName',
  photoUrl: 'photoUrl',
  googlePhotoUrl: 'googlePhotoUrl',
  themeColor: 'themeColor',
  themeMode: 'themeMode',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
});

exports.Prisma.TransactionScalarFieldEnum = makeEnum({
  id: 'id',
  uid: 'uid',
  categoryId: 'categoryId',
  comment: 'comment',
  integerAmount: 'integerAmount',
  time: 'time',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  scheduleId: 'scheduleId'
});

exports.Prisma.TransactionScheduleScalarFieldEnum = makeEnum({
  id: 'id',
  uid: 'uid',
  categoryId: 'categoryId',
  comment: 'comment',
  integerAmount: 'integerAmount',
  intervalType: 'intervalType',
  intervalEvery: 'intervalEvery',
  firstOccurrence: 'firstOccurrence',
  occurrences: 'occurrences',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  latestCreatedOccurrence: 'latestCreatedOccurrence'
});

exports.Prisma.BudgetScalarFieldEnum = makeEnum({
  id: 'id',
  uid: 'uid',
  label: 'label',
  integerAmount: 'integerAmount',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt',
  periodMonths: 'periodMonths'
});

exports.Prisma.BudgetCategoryInclusionScalarFieldEnum = makeEnum({
  budgetId: 'budgetId',
  categoryId: 'categoryId'
});

exports.Prisma.CategoryScalarFieldEnum = makeEnum({
  id: 'id',
  uid: 'uid',
  value: 'value',
  icon: 'icon',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
});

exports.Prisma.FeedbackScalarFieldEnum = makeEnum({
  id: 'id',
  uid: 'uid',
  message: 'message'
});

exports.Prisma.LogScalarFieldEnum = makeEnum({
  id: 'id',
  uid: 'uid',
  type: 'type',
  message: 'message',
  data: 'data',
  stackTrace: 'stackTrace',
  device: 'device',
  timestamp: 'timestamp',
  timestring: 'timestring',
  createdAt: 'createdAt',
  href: 'href'
});

exports.Prisma.ConfigScalarFieldEnum = makeEnum({
  key: 'key',
  value: 'value'
});

exports.Prisma.PremiumSubscriptionScalarFieldEnum = makeEnum({
  id: 'id',
  uid: 'uid',
  currentPeriodEnd: 'currentPeriodEnd',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
});

exports.Prisma.PremiumPriceScalarFieldEnum = makeEnum({
  id: 'id',
  unitAmount: 'unitAmount',
  productId: 'productId',
  active: 'active',
  currency: 'currency',
  nickname: 'nickname',
  type: 'type',
  recurringInterval: 'recurringInterval',
  recurringIntervalCount: 'recurringIntervalCount',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
});

exports.Prisma.SortOrder = makeEnum({
  asc: 'asc',
  desc: 'desc'
});

exports.Prisma.QueryMode = makeEnum({
  default: 'default',
  insensitive: 'insensitive'
});
exports.IntervalType = makeEnum({
  DAY: 'DAY',
  WEEK: 'WEEK',
  MONTH: 'MONTH',
  YEAR: 'YEAR'
});

exports.Prisma.ModelName = makeEnum({
  User: 'User',
  Profile: 'Profile',
  Transaction: 'Transaction',
  TransactionSchedule: 'TransactionSchedule',
  Budget: 'Budget',
  BudgetCategoryInclusion: 'BudgetCategoryInclusion',
  Category: 'Category',
  Feedback: 'Feedback',
  Log: 'Log',
  Config: 'Config',
  PremiumSubscription: 'PremiumSubscription',
  PremiumPrice: 'PremiumPrice'
});

/**
 * Create the Client
 */
class PrismaClient {
  constructor() {
    throw new Error(
      `PrismaClient is unable to be run in the browser.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues`,
    )
  }
}
exports.PrismaClient = PrismaClient

Object.assign(exports, Prisma)
